### DISCUSSION TITLE
Is it possible to specify a commit SHA when fetching/building?

### DISCUSSION THREAD

**Discussion by mjram0s** [coin-or/coinbrew#77] [Q&A] [answered]
Hello. I want to specify a specific commit (not version or master) to build. For example `coinbrew build Cbc@<commit SHA>`. Is this possible? I don't see it mentioned in the docs.

**Comment by tkralphs** [comment, marked as answer]
Yes, it's possible, but the mechanism should be improved. There seems to be a small bug. With the current version, it seems you have to do
```
coinbrew fetch Cbc@master --main-proj-sha=<commit SHA> 
```
The request for `master` ends up being ignored, but if you don't specify it, coinbrew will complain.

Note, however, that this does not check out the dependencies as they were at that time, so what you probably really want is to use the `--time` option, which checks out all packages at a particular point in time. You can use the date format of the git commit log. So do something like
```
~/projects/Clp> git show c5b57d3d759a6c
commit c5b57d3d759a6cf198600d768df87761e65e406c
Author: Ruan Luies <omega@live.co.za>
Date:   Mon Jul 25 17:21:10 2022 +0200
...
~/Projects> coinbrew fetch Clp@master --time="Mon Jul 25 17:21:10 2022"
```
Note that the `--time` option has the same bug. If you open an issue on this, I'll try to have a look at it.

**Reply by tkralphs** [reply]
I guess it would be be easy to make the mechanism for checking out a particular SHA also figure out the time at which to check out other dependencies. I can also try to look at that if you open an issue.

**Reply by mjram0s** [reply]
Thank you for your help, I'll try the `--time` option. I opened a new issue from your comment and will add to it if I run into problems.

**Reply by mjram0s** [reply]
When building after fetching, do I just need to pass in the project name? e.g.
```
> coinbrew fetch Clp@master --time="Mon Jul 25 17:21:10 2022"
> coinbrew build Clp
```

**Reply by tkralphs** [reply]
Yes, if you don't specify version, whatever version is already fetched will be built. You can actually do building and fetching all at once. If you specify a version with the build command that has not already been fetched, it will automatically fetch it first. I personally prefer just to do the fetch first and then build.
